service.php

```diff
@@ -317,6 +317,19 @@ function token_select_box( $name, $create = false ) {
 		$tokens = $this->get_tokens();
 		return Keyring_Util::token_select_box( $tokens, $name, $create );
 	}
+
+	/**
+	 * A shortcut to query registered endpoints.  It saves having to re-extract the url and method manually.
+	 *
+	 * @param string $endpoint The registered endpoint (via `->set_endpoint()`) to query.
+	 * @return Keyring_Error|String Either an error or the response data.
+	 */
+	function request_endpoint( $endpoint ) {
```

Comment:
This helper method is currently only used by the LinkedIn service on wpcom, so have included with this PR.

Reply:
Interestingly, we don't actually use this in `Keyring_Service_LinkedIn`, but a subclass of it.

I think it's a useful shortcut, method however. It's a very common use case to call `$this->request( $this->self_url, array( 'method' => $this->self_method ) )`, or something of that nature.

To make it more generally useful, perhaps the `::request_endpoint()` method should take an optional `$params` argument, to which it would add `$params['method'] = $this->{"{$endpoint}_method"};`, then pass to `$this->request()`.

Reply:
Done